libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt

```diff
@@ -1,29 +1,413 @@
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_ROUNDABOUT_TURN
+import com.mapbox.navigation.base.model.route.RouteConstants.STEP_MANEUVER_TYPE_TURN
+import com.mapbox.navigation.base.model.route.RouteProgress
+import com.mapbox.navigation.base.options.TripNavigationOptions
 import com.mapbox.navigation.base.trip.TripNotification
+import com.mapbox.navigation.utils.extensions.inferDeviceLanguage
+import com.mapbox.navigation.utils.extensions.inferDeviceLocale
+import com.mapbox.navigation.utils.formatter.distance.DistanceFormatter
+import com.mapbox.navigation.utils.formatter.distance.getUnitTypeForLocale
+import com.mapbox.navigation.utils.formatter.time.TimeFormatter.formatTime
+import java.util.Calendar
 
-class MapboxTripNotification : TripNotification {
-    override fun getNotification(): Notification {
-        TODO("not implemented")
+class MapboxTripNotification(
+    private val applicationContext: Context,
+    private val tripNavigationOptions: TripNavigationOptions,
+    private var currentRoute: Route
+) : TripNotification {
+
+    private val END_NAVIGATION_ACTION = "com.mapbox.intent.action.END_NAVIGATION"
+    private val SET_BACKGROUND_COLOR = "setBackgroundColor"
+
+    private var notificationManager: NotificationManager? = null
+    private lateinit var notification: Notification
+    private var collapsedNotificationRemoteViews: RemoteViews? = null
+    private var expandedNotificationRemoteViews: RemoteViews? = null
+    private var currentDistanceText: SpannableString? = null
+    private var instructionText: String? = null
+    private var currentManeuverId: Int = 0
+    private var isTwentyFourHourFormat: Boolean = false
+    private var etaFormat: String = ""
+    private var pendingOpenIntent: PendingIntent? = null
+    private var pendingCloseIntent: PendingIntent? = null
+    private var distanceFormatter: DistanceFormatter? = null
+
+    private val endNavigationBtnReceiver = object : BroadcastReceiver() {
+        override fun onReceive(applicationContext: Context, intent: Intent) {
+            // this@MapboxNavigationNotification.onEndNavigationBtnClick()
+        }
     }
 
-    override fun getNotificationId(): Int {
-        TODO("not implemented")
+    companion object {
+        private const val NAVIGATION_NOTIFICATION_ID = 5678
+        private const val NAVIGATION_NOTIFICATION_CHANNEL = "NAVIGATION_NOTIFICATION_CHANNEL"
+    }
+
+    init {
+        etaFormat = applicationContext.getString(R.string.eta_format)
+        initializeDistanceFormatter()
+        notificationManager = applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as? NotificationManager
+        isTwentyFourHourFormat = DateFormat.is24HourFormat(applicationContext)
+
+        pendingOpenIntent = createPendingOpenIntent()
+        pendingCloseIntent = createPendingCloseIntent()
+
+        registerReceiver()
+        createNotificationChannel()
+        if (!::notification.isInitialized) {
```

Comment:
We can make `notification` just `private var`, without `lateinit` -> we can remove this check